cloud/cloud_env_impl.h

```diff
@@ -237,7 +232,6 @@ class CloudEnvImpl : public CloudEnv {
     return base_env_->GetFreeSpace(path, diskfree);
   }
 
-  CloudManifest* GetCloudManifest() { return cloud_manifest_.get(); }
```

Comment:
This function was public before the change. I'm trying to limit access to `cloud_manifest_` since it will be protected by shared lock if we want to support switching it without reopening db.

After making it private, I felt there is no point in keeping this function anymore so I simply removed it.